wallet/src/Pos/Wallet/Web/Account.hs

```diff
@@ -132,14 +132,13 @@ generateUnique desc RandomSeed generator isDuplicate = loop (100 :: Int)
                       \you are approaching the limit") desc
     loop i = do
         rand  <- liftIO randomIO
-        value <- generator rand
-        bad   <- orM
-            [ isDuplicate rand value
-            , pure $ isHardened rand  -- using hardened keys only for now
-            ]
-        if bad
-            then loop (i - 1)
-            else return value
+        if isHardened rand then
+            loop (i - 1) -- use non-hardened keys only for now
+        else do
+            value <- generator rand
+            isDup <- isDuplicate rand value
```

Comment:
@pva701 Have you tried speeding up the underlying function which `isDuplicate` uses as described [here](https://iohk.myjetbrains.com/youtrack/issue/CSL-2120#comment=93-16616)? Do you notice any substantial improvement or was I being mental there?